src/celeritas/neutron/data/NeutronElasticData.hh

```diff
@@ -0,0 +1,189 @@
+
+    // Coefficients
+    ChipsArray par{};  //!< Coefficients as a function of atomic mass numbers
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Model and particles IDs for neutron--nucleus elastic scattering.
+ */
+struct NeutronElasticIds
+{
+    ActionId action;
+    ParticleId neutron;
+
+    //! Whether the IDs are assigned
+    explicit CELER_FUNCTION operator bool() const { return action && neutron; }
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Microscopic cross section data for the neutron elastic interaction.
+ */
+template<Ownership W, MemSpace M>
+struct NeutronElasticXsData
+{
+    using XsUnits = units::Native;  // [len^2]
+
+    template<class T>
+    using Items = Collection<T, W, M>;
+    template<class T>
+    using ElementItems = Collection<T, W, M, ElementId>;
+
+    //// MEMBER DATA ////
+
+    Items<real_type> reals;
+    ElementItems<GenericGridData> elements;
+
+    //// MEMBER FUNCTIONS ////
+
+    //! Whether all data are assigned and valid
+    explicit CELER_FUNCTION operator bool() const
+    {
+        return !reals.empty() && !elements.empty();
+    }
+
+    //! Assign from another set of data
+    template<Ownership W2, MemSpace M2>
+    NeutronElasticXsData& operator=(NeutronElasticXsData<W2, M2> const& other)
+    {
+        CELER_EXPECT(other);
+        reals = other.reals;
+        elements = other.elements;
+        return *this;
+    }
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * A-dependent data for the differential cross section (momentum transfer) of
+ * the CHIPS neutron-nucleus elastic model.
+ */
+template<Ownership W, MemSpace M>
+struct NeutronElasticDiffXsData
+{
+    template<class T>
+    using IsotopeItems = Collection<T, W, M, IsotopeId>;
+
+    //// MEMBER DATA ////
+
+    IsotopeItems<ChipsDiffXsCoefficients> coeffs;
+
+    //// MEMBER FUNCTIONS ////
+
+    //! Whether all data are assigned and valid
+    explicit CELER_FUNCTION operator bool() const
+    {
+        return !coeffs.empty() && !coeffs.empty();
+    }
+
+    //! Assign from another set of data
+    template<Ownership W2, MemSpace M2>
+    NeutronElasticDiffXsData&
+    operator=(NeutronElasticDiffXsData<W2, M2> const& other)
+    {
+        CELER_EXPECT(other);
+        coeffs = other.coeffs;
+        return *this;
+    }
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Device data for creating an interactor.
+ */
+template<Ownership W, MemSpace M>
+struct NeutronElasticData
+{
+    //! Model and particle IDs
+    NeutronElasticIds ids;
+
+    //! Particle mass * c^2 [MeV]
+    units::MevMass neutron_mass;
+
+    //! Neutron elastic cross section (G4PARTICLEXS/neutron/elZ) data
+    NeutronElasticXsData<W, M> xs;
+
+    //! Neutron elastic differential cross section coefficients
+    NeutronElasticDiffXsData<W, M> Q2;
```

Comment:
Use words, not math symbols
```suggestion
    NeutronElasticDiffXsData<W, M> diff_xs;
```

Reply:
Okay, I have a big bias for Q2 (as HEP conventionally uses the capital Q^2 for the momentum transfer (in any virtual particle or quark/gluon exchanges.  I almost used q2 instead, but have failed to convince myself.  Not a big deal, so I changed it to diff_xs and  Q2->q2 in other places - I would like to keep q2 instead of q_square in most places if this is still acceptable for our coding style.

Reply:
Just gonna leave this star trek reference here...

![image](https://github.com/celeritas-project/celeritas/assets/741229/8234ad2c-3695-4904-8f6a-15804c36c55c)

Reply:
Well, it should be $Q^2$, not $Q2$. And Amanda `Lund`, not Rogers - sorry that I can not fully digest this joke(?) anyway.

Reply:
nor can I, star trek references are usually over my head ;)

Reply:
Not really a joke to be had except to share the bizarre associations my brain makes due to my high school nerdiness.